src/inmanta/agent/cache.py

```diff
@@ -132,70 +139,37 @@ def __init__(self, agent_instance: Optional["AgentInstance"] = None) -> None:
         self.addLocks: dict[str, Lock] = {}
         self._agent_instance = agent_instance
 
+        # Set of lingering cache items used during a resource action.
+        self.lingering_set: set[CacheItem] = set()
+
+    def touch_used_cache_items(self) -> None:
+        """
+        Extend the expiry time of items in the lingering_set by 60s
+        """
+        now = time.time()
+        for item in self.lingering_set:
+            item.expiry_time = now + 60
+
+        self.lingering_set = set()
+        if self.timer_queue:
+            # Bad O(N) but unavoidable (?) Since we modify elements above
+            # we have to make sure the heap is still a heap.
+            heapq.heapify(self.timer_queue)
+            self.next_action = self.timer_queue[0].expiry_time
```

Comment:
I think this is pretty bad but I don't see a way around it.

Reply:
N is likely to be small, so I would leave it at this.